src/LibVLCSharp.Forms/Shared/PlaybackControls.xaml.cs

```diff
@@ -821,14 +859,126 @@ private void OnAudioTracksChanged(object sender, EventArgs e)
             UpdateAudioTracksSelectionAvailability();
         }
 
+        private void OnVideoTracksChanged(object sender, EventArgs e)
+        {
+            UpdateVideoTracksSelectionAvailability();
+        }
+
         private void OnSubtitlesTracksChanged(object sender, EventArgs e)
         {
             UpdateClosedCaptionsTracksSelectionAvailability();
         }
 
-        private async void AudioTracksSelectionButton_ClickedAsync(object sender, EventArgs e)
+
+        private void TracksButton_Clicked(object sender, EventArgs e)
+        {
+            if (ControlsPanel != null)
+                ControlsPanel.IsVisible = false;
+            var audioTrackSouces = LoadTracks(Manager.Get<AudioTracksManager>());
+            var videoTrackSouces = LoadTracks(Manager.Get<VideoTracksManager>());
+            var subtitleTracksSources = LoadTracks(Manager.Get<SubtitlesTracksManager>());
+            if (AudioTracksListView != null && VideoTracksListView != null && SubtitlesTracksListView != null)
+            {
+                AudioTracksListView.ItemsSource = audioTrackSouces;
+                VideoTracksListView.ItemsSource = videoTrackSouces;
+                SubtitlesTracksListView.ItemsSource = subtitleTracksSources;
+            }
+
+            if (TracksOverLayView != null)
+                TracksOverLayView.IsVisible = true;
+        }
+
+        private void AudioTracksItemTapped(object sender, ItemTappedEventArgs e)
+        {
+            var track = (TrackDescription)e.Item;
+            var manager = Manager.Get<AudioTracksManager>();
+            SelectTrack(manager, track, AudioTracksListView);
+        }
+
+        private void VideoTracksItemTapped(object sender, ItemTappedEventArgs e)
         {
-            await SelectTrackAsync(Manager.Get<AudioTracksManager>(), ResourceManager.GetString(nameof(Strings.AudioTracks)));
+            var track = (TrackDescription)e.Item;
+            var manager = Manager.Get<VideoTracksManager>();
+            SelectTrack(manager, track, VideoTracksListView);
+        }
+
+        private void SubtitlesTracksItemTapped(object sender, ItemTappedEventArgs e)
+        {
+            var track = (TrackDescription)e.Item;
+            var manager = Manager.Get<SubtitlesTracksManager>();
+            SelectTrack(manager, track, SubtitlesTracksListView);
+        }
+
+        private void SelectTrack(TracksManager manager, TrackDescription track, ListView? tracksListview)
+        {
+            var tracks = manager.Tracks;
+            var currentTrackId = manager.CurrentTrackId;
+            if (tracks == null || track.Name.Contains("*"))
+                return;
+            var found = false;
+            var index = 0;
+            foreach (var trackDescription in tracks)
+            {
+                index += 1;
+                if (GetTrackName(trackDescription, currentTrackId) == track.Name)
```

Comment:
I don't get why you compare on the name, where `*` is appended if it is the current track... Why not using the Track Id ?

Reply:
The trackDescription Ids are not are not ordered like 0,1,2  so that I can index my Itemsource.
For VideoTrack IDs for example, you can see 0, -1, 3.
So when creating the Itemsource for a ListView, I change the index: -1 become 0, -1 become 1, ...
That is why I make the comparison using this name.
You will notice that currently in the 3.x branch, the Track Id is used for the comparaion and it's wrong.